BLSQ/iaso
# IA-3468 IA-3511: Sentry front-end

Implement Sentry front-end solution on Dashboard, using replays to catch errors and for user tracking sessions

Related JIRA tickets : IA-3468  IA-3511

## Self proofreading checklist


## Doc
You need those Env variables to make sentry work:
```

SENTRY_ENVIRONMENT=ENV_NAME
SENTRY_URL=URL_FROM_AWS_ENV
```

## Changes

- custom provider to initialise Sentry and show a dialog warning the user 
- Adapt `fetchIaso` to trigger a Sentry error.
- adapt protected routes to set current user to sentry

## How to test
Deploy this branch on staging.
Try to create a new user using same username as another one.
Close the window.
Visit sentry replays, you should have a new replay with your action


## Print screen / video

<img width="1676" alt="Screenshot 2024-10-29 at 22 55 38" src="https://github.com/user-attachments/assets/483c6b33-4ea0-4d53-990b-cfbd34a6c6c5">
